### PR TITLE
Deprecate 2 bad methods

### PR DESCRIPTION
**Pull request by gst**
2 (obvious) methods in less.

NB: The test will fail because of https://github.com/Alignak-monitoring/alignak/issues/164

### CODE REVIEW

File: alignak/basemodule.py [hunk 1]
```diff
@@ -273,6 +274,11 @@ def has(self, prop):
         :return: True if has a property, otherwise False
         :rtype: bool
         """
+        warnings.warn(
```
Comment (ddurieux) [inline alignak/basemodule.py]: Why not use logger.warning?

**Comment by ddurieux** [comment]
There are too many warn in the tests, perhaps fix them before merge ;)

File: alignak/basemodule.py (hunk 1 above)
Comment (gst) [inline alignak/basemodule.py]: À priori, this is the python way for deprecating ;)

**Comment by gst** [comment]
? Related to this change ?

File: alignak/basemodule.py (hunk 1 above)
Comment (ddurieux) [inline alignak/basemodule.py]: Ok ;)

**Comment by ddurieux** [comment]
Yes, many:

/home/travis/build/Alignak-monitoring/alignak/alignak/objects/itemgroup.py:99: DeprecationWarning: Servicegroup is deprecated, please use `hasattr(your_object, attr)` instead. This has() method will be removed in a later version.

and on other files

**Comment by Seb-Solon** [comment]
Imo, you don't need 5 commits for that simple replacement of has in hasattr.

Anyway you should rebase, as test a now working.

**Comment by gst** [comment]
@ddurieux  : sweet, that's what a/the deprecation warning is supposed to do :p 

I'll give a look to remove/update the others usages of those 2 functions.

@Seb-Solon :  that's how deprecation is supposed to happen : first you emit/add a warning in the code you want to deprecate, then - and it could be done in more than one supplementary step - you remove/update the usages of the deprecated thing (possibly you may not found all the usages directly, so the need/usefull of the warning)  ;)
but I could have done all of it in 1 commit (per deprecated function/thing). it's just better like it is :p

and yeah I know I'll rebase :p but somehow later..

**Comment by Seb-Solon** [comment]
>: that's how deprecation is supposed to happen : first you emit/add a warning in the code you want to deprecate, then - and it could be done in more than one supplementary step - you remove/update the usages of the deprecated thing (possibly you may not found all the usages directly, so the need/usefull of the warning) ;)

I was not talking about deprecation. +27 −6 could be a single commit, that's all.

**Comment by gst** [comment]
here it is..

**Comment by gst** [comment]
pep8 fixed

**Comment by gst** [comment]
rebased, should be super green now..

**Comment by ddurieux** [comment]
I'm ok this this, perhaps squash it in only one commit

**Comment by gst** [comment]
not necessary to squash, that's not forth and back commit with dirty changes done and undone ;)

**Comment by ddurieux** [comment]
Ok, so you have do all changes needed?  if yes I merge

**Comment by gst** [comment]
yes,

if I'd have forgotten (not impossible though I don't think there are) to
replace some `has(..)` calls then they will be catched by the deprecation
warning (and still be correctly run).

so .. go :p

2015-08-31 8:58 GMT-04:00 David Durieux <notifications@github.com>:

**Pull request merged**